src/screens/WebsocketResearchScreen/SignedChatScreen.tsx

```diff
@@ -38,52 +50,58 @@ const SignedChatScreen = () => {
-        <ChatDetailHeader
-          channel={selectedChannel}
-          onAvatarPress={goToChatInfoPage}
-          onBackPress={goBackFromChatScreen}
-          onThreeDotPress={goToChatInfoPage}
-          avatar={selectedChannel?.channelPicture}
-          type={SIGNED}
-          user={
-            selectedChannel?.rawJson?.channel?.anon_user_info_emoji_code
-              ? `Anonymous ${selectedChannel?.rawJson?.channel?.anon_user_info_emoji_name} `
-              : getChatName(selectedChannel?.name, profile?.myProfile?.username)
-          }
-          anon_user_info_emoji_code={selectedChannel?.rawJson?.channel?.anon_user_info_emoji_code}
-          anon_user_info_color_code={selectedChannel?.rawJson?.channel?.anon_user_info_color_code}
+    <ScrollContext.Provider value={scrollContext}>
+      <View style={styles.keyboardAvoidingView}>
+        {selectedChannel ? (
+          <ChatDetailHeader
+            channel={selectedChannel}
+            onAvatarPress={goToChatInfoPage}
+            onBackPress={goBackFromChatScreen}
+            onThreeDotPress={goToChatInfoPage}
+            avatar={selectedChannel?.channelPicture}
+            type={SIGNED}
+            user={
+              selectedChannel?.rawJson?.channel?.anon_user_info_emoji_code
+                ? `Anonymous ${selectedChannel?.rawJson?.channel?.anon_user_info_emoji_name} `
+                : getChatName(selectedChannel?.name, profile?.myProfile?.username)
+            }
+            anon_user_info_emoji_code={selectedChannel?.rawJson?.channel?.anon_user_info_emoji_code}
+            anon_user_info_color_code={selectedChannel?.rawJson?.channel?.anon_user_info_color_code}
+          />
+        ) : null}
+
+        <FlatList
+          ref={scrollRef}
+          contentContainerStyle={styles.flatlistContainer}
+          style={styles.chatContainer}
+          data={updatedChats}
+          inverted={true}
+          initialNumToRender={20}
+          alwaysBounceVertical={false}
+          keyExtractor={(item, index) => item?.id || index.toString()}
+          renderItem={renderChatItem}
         />
-      ) : null}
 
-      <FlatList
-        contentContainerStyle={styles.contentContainerStyle}
-        style={styles.chatContainer}
-        data={updatedChats}
-        inverted={true}
-        initialNumToRender={20}
-        alwaysBounceVertical={false}
-        bounces={false}
-        onLayout={scrollToEnd}
-        keyExtractor={(item, index) => item?.id || index.toString()}
-        renderItem={renderChatItem}
-      />
-      <View style={styles.inputContainer}>
-        <AnonymousInputMessage onSendButtonClicked={sendChat} type={SIGNED} />
+        {replyPreview && <ChatReplyPreview type={SIGNED} />}
+        <View style={styles.inputContainer}>
+          <AnonymousInputMessage onSendButtonClicked={sendChat} type={SIGNED} />
+        </View>
+        <Loading visible={loading} />
       </View>
-    </View>
+    </ScrollContext.Provider>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the new hunk, the `scrollToEnd` functionality seems to be missing. If this functionality is still required, consider adding it back in. Also, the `bounces={false}` prop has been removed from the `FlatList` component. If you want to disable bouncing completely for the list, you should add it back.
```diff
  <FlatList
    ref={scrollRef}
    contentContainerStyle={styles.flatlistContainer}
    style={styles.chatContainer}
    data={updatedChats}
    inverted={true}
    initialNumToRender={20}
    alwaysBounceVertical={false}
+   bounces={false}
    keyExtractor={(item, index) => item?.id || index.toString()}
    renderItem={renderChatItem}
  />
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->